apps/protoc-gen-openapi/main.go

```diff
@@ -16,13 +16,27 @@
 package main
 
 import (
-	"google.golang.org/protobuf/compiler/protogen"
+	"flag"
 
 	"github.com/google/gnostic/apps/protoc-gen-openapi/generator"
+	"google.golang.org/protobuf/compiler/protogen"
 )
 
+var flags flag.FlagSet
+
 func main() {
-	protogen.Options{}.Run(func(plugin *protogen.Plugin) error {
-		return generator.NewOpenAPIv3Generator(plugin).Run()
+	conf := generator.Configuration{
+		Version:     flags.String("version", "0.0.1", "version number text, e.g. 1.2.3"),
+		Title:       flags.String("title", "", "name of the API"),
+		Description: flags.String("description", "", "description of the API"),
+		JSONNames:   flags.Bool("json", false, "use JSON naming convention (camelCase)"),
```

Comment:
Could we rename this to "Naming" and make it a string argument? I think it's possible that we could want to support other naming conventions explicitly (possibly involving capitalization of various things) and a string argument is more extensible than a boolean.

We could also discuss what we should use for the value of this flag and what its default should be.

I did a quick search for "JSON naming convention"; the top hits were:

- https://stackoverflow.com/questions/5543490/json-naming-convention-snake-case-camelcase-or-pascalcase
- https://google.github.io/styleguide/jsoncstyleguide.xml
- https://newbedev.com/json-naming-convention-snake-case-camelcase-or-pascalcase

Reading the comments on the first post, there doesn't seem to be a standard "JSON" style, but the second and third cite the Google JSON Style Guide. Since we're mainly in a Google-led ecosystem, I think it would be appropriate to make this the default (and use "proto" or "raw" for our previous unprocessed names). 

I'd like to check to see which form of these names is supported by Envoy's gRPC transcoding. (I'll test something and follow up)

Reply:
I found that for my test API (https://github.com/apigee/registry), the Envoy transcoder accepted both camel case and snake case field names in posted JSON and always returned camel case field names (regardless of the format that I used when sending values). So I think camel case should be the default. (We don't have to change this in this PR but I would want to do that soon)

Reply:
Thank you! I didn't really like the name "JSONNames", but was apparently to tired to come up with something better.
It makes much more sense to make it "Naming" and open for more conventions in the future 👍 

If remember correctly, it should be easy to change default to camel case.

Reply:
As a proposal, how about calling the flag "naming" and using "json" for the Google JSON style, making that the default, and "proto" for the original naming, which is really just taking names from the protos as-is.

Reply:
Renamed in latest pull request to "naming", with "json" as default value and "proto" as the second option.